fairlearn/metrics/_metric_frame.py

```diff
@@ -147,8 +288,9 @@ def __init__(self,
                 raise ValueError(_DUPLICATE_FEATURE_NAME.format(name))
             nameset.add(name)
 
-        self._overall = self._compute_overall(func_dict, y_t, y_p, cf_list)
-        self._by_group = self._compute_by_group(func_dict, y_t, y_p, sf_list, cf_list)
+        self.func_dict = self._process_functions(self.metric, s_p)
```

Comment:
Why has the processing of the functions been moved out of the constructor?

Reply:
Because the process_functions is dependent on the sample_params.